src/pages/settings/ProfilePage.js

```diff
@@ -1,20 +1,271 @@
+import {setPersonalDetails} from '../../libs/actions/PersonalDetails';
 import ROUTES from '../../ROUTES';
+import ONYXKEYS from '../../ONYXKEYS';
+import CONST from '../../CONST';
+import Avatar from '../../components/Avatar';
+import styles from '../../styles/styles';
+import Text from '../../components/Text';
+import {DownArrow} from '../../components/Icon/Expensicons';
+import Icon from '../../components/Icon';
+import Checkbox from '../../components/Checkbox';
+import themeColors from '../../styles/themes/default';
 
-const ProfilePage = () => (
-    <ScreenWrapper>
-        <HeaderWithCloseButton
-            title="Profile"
-            shouldShowBackButton
-            onBackButtonPress={() => Navigation.navigate(ROUTES.SETTINGS)}
-            onCloseButtonPress={() => Navigation.dismissModal()}
-        />
-    </ScreenWrapper>
-);
+const propTypes = {
+    /* Onyx Props */
+    // The personal details of the person who is logged in
+    myPersonalDetails: PropTypes.shape({
+        // Email/Phone login of the current user from their personal details
+        login: PropTypes.string,
 
+        // Display first name of the current user from their personal details
+        firstName: PropTypes.string,
+
+        // Display last name of the current user from their personal details
+        lastName: PropTypes.string,
+
+        // Avatar URL of the current user from their personal details
+        avatar: PropTypes.string,
+
+        // Pronouns of the current user from their personal details
+        pronouns: PropTypes.string,
+
+        // timezone of the current user from their personal details
+        timezone: PropTypes.shape({
+
+            // Value of selected timezone
+            selected: PropTypes.string,
+
+            // Whether timezone is automatically set
+            automatic: PropTypes.bool,
+        }),
+    }),
+};
+
+const defaultProps = {
+    myPersonalDetails: {},
+};
+
+const timezones = moment.tz.names()
+    .map(timezone => ({
+        value: timezone,
+        label: timezone,
+    }));
+
+class ProfilePage extends Component {
+    constructor(props) {
+        super(props);
+
+        const {
+            firstName,
+            lastName,
+            pronouns,
+            timezone = {},
+        } = props.myPersonalDetails;
+        const pronounsList = Object.values(CONST.PRONOUNS);
+
+        let currentUserPronouns = pronouns;
+        let initialSelfSelectedPronouns = '';
+
+        // This handles populating the self-selected pronouns in the form
+        if (pronouns && !pronounsList.includes(pronouns)) {
+            currentUserPronouns = CONST.PRONOUNS.SELF_SELECT;
+            initialSelfSelectedPronouns = pronouns;
+        }
+
+        this.state = {
+            firstName,
+            lastName,
+            pronouns: currentUserPronouns,
+            selfSelectedPronouns: initialSelfSelectedPronouns,
+            selectedTimezone: timezone.selected ?? CONST.DEFAULT_TIME_ZONE.selected,
```

Comment:
Sorry for not catching this earlier, but I believe prefer to use `||` over `??` when possible